tests/e2e/runtime.go

```diff
@@ -0,0 +1,465 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"time"
+
+	"google.golang.org/grpc"
+
+	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	"github.com/oasisprotocol/oasis-core/go/common"
+	coreSig "github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+	cmnGrpc "github.com/oasisprotocol/oasis-core/go/common/grpc"
+	"github.com/oasisprotocol/oasis-core/go/common/node"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/cmd"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/env"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/log"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/oasis"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/scenario"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/scenario/e2e"
+	registry "github.com/oasisprotocol/oasis-core/go/registry/api"
+	scheduler "github.com/oasisprotocol/oasis-core/go/scheduler/api"
+	"github.com/oasisprotocol/oasis-core/go/storage/database"
+
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/client"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+)
+
+const (
+	cfgRuntimeBinaryDirDefault = "runtime.binary_dir.default"
+	cfgRuntimeLoader           = "runtime.loader"
+	cfgIasMock                 = "ias.mock"
+	cfgEpochInterval           = "epoch.interval"
+)
+
+var (
+	// RuntimeParamsDummy is a dummy instance of runtimeImpl used to register global e2e/runtime flags.
+	RuntimeParamsDummy *runtimeImpl = newRuntimeImpl("", []RunTestFunction{})
+
+	// SimpleKVRuntime is the basic network + client test case with runtime support.
+	SimpleKVRuntime scenario.Scenario = newRuntimeImpl("simple-keyvalue-runtime", []RunTestFunction{SimpleKVTest})
```

Comment:
You should really split these things up, so you define the common runtime runner in one file and then the key-value runtime scenario in a separate file etc.

Reply:
Yeah, this is an early draft, I'll do the cosmetic fixes after I get it working :)